### PR TITLE
Add Node.js 12 to CI Services

### PR DESCRIPTION
**Pull request by erisu**
### Platforms affected
none

### Motivation and Context
Add Node 12 too CI

### Description
- Updated CI Service Config Files

### Testing
none

### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Comment by erisu** [comment]
@timbru31 Without quotes doesn't cause any issues but adding it isn't a problem.

YAML in general does not require quotation. There are exceptions for example the numbers wanted to be interpreted as string instead of a number, these should be explicitly wrapped.

Since this value is piped into a terminal command (`NVM`) NVM will take the value as a string.

```
nvm use 10.0.0
```

Here is also an example of a string value for `node_js` that they forgot to wrap with quotes in the doc.
https://docs.travis-ci.com/user/build-environment-updates/2015-02-03/#iojs-support

For Elixir, they even talked about converting int/float values to strings only for consistency of the doc, not due to issues of the value not as a string. https://github.com/travis-ci/docs-travis-ci-com/pull/1490#issuecomment-335889711

File: appveyor.yml [hunk 1]
```diff
@@ -22,9 +22,14 @@
...
-  - nodejs_version: "6"
-  - nodejs_version: "8"
-  - nodejs_version: "10"
+  - nodejs_version: 6
+  - nodejs_version: 8
+  - nodejs_version: 10
+  - nodejs_version: 12
+
+matrix:
+  allow_failures:
```
Comment (janpio) [inline appveyor.yml]: I still don't get why this is useful.
Why not just leave it out for AppVeyor?

Comment (erisu) [reply appveyor.yml]: There is actually a work around. 

We could just use this instead to install any node version we want. This should allow us to install any Node.js version including the ones that are not on AppVeyor's image.

```
ps: Update-NodeJsInstallation (Get-NodeJsLatestBuild $env:nodejs_version) $env:platform
```

https://www.appveyor.com/docs/lang/nodejs-iojs/#installing-any-version-of-nodejs-or-iojs

Comment (janpio) [reply appveyor.yml]: Wouldn't it make the installs/buils for the existing versions slower than necessary?

**Pull request merged**